src/performance.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * This file is part of the Game Closure SDK.
+ *
+ * The Game Closure SDK is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public License v. 2.0 as published by Mozilla.
+
+ * The Game Closure SDK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public License v. 2.0 for more details.
+
+ * You should have received a copy of the Mozilla Public License v. 2.0
+ * along with the Game Closure SDK.  If not, see <http://mozilla.org/MPL/2.0/>.
+ */
+
+ /**
+ * @module performanceMetric
+ *
+ * Measures performance by keeping a window of delta times and providing a score using that window.
+ *
+**/
+
+var engine = null;
+
+var DEFAULT_RANK = 0;
+var DEFAULT_ALLOW_REDUCTION = true;
+var LOWER_BOUND = 10;
+var HIGHER_BOUND = 60;
+var TICKS_TIL_INCREASE_SCORE = 50;
+  
+var Performance = Class(function () {
+  var ticksSinceLastWorstUpdate = 0;
+  var worstTicksSize = 30;
+  var worstTicks = [];
+  var minFPS = LOWER_BOUND;
+  var maxFPS = HIGHER_BOUND;
+  var lastTick;
+
+  var _map = function(val, start1, stop1, start2, stop2) {
```

Comment:
This function is really vague - as people look at this file for the first time, it won't make any sense on first read.

Let's try `var _mapFPSToPerformanceScore = function (fps) {`

The other parameters are all private variables or constants and don't need to be passed in